### DISCUSSION TITLE
Adding a maintainer?

### DISCUSSION THREAD

**Discussion by jacksongoode** [jpochyla/psst#359] [Q&A]
Psst is currently the closest, cross-platform alternative to the Spotify client. I see a bunch of great work from the community in fixing bugs and adding features but understand that @jpochyla has been pretty busy with a new kid and job.

I'm wondering if you have thought about adding another maintainer? Someone who has worked with other Rust UI projects (and druid) might be able to help shepherd the work the community is doing into new builds.

Is anyone interested in this role (if @jpochyla would be interested first of course)? Just pinging a few folks who have made repeated recent contributions into this discussion. @gordonshieh @iandwelker @swiesmann @Insprill

**Comment by Insprill** [comment]
If jpochyla is interested I would definitely be willing to help out!

**Comment by jacksongoode** [comment]
@jpochyla Just in case the original ping was missed (I promise no more pings after this :) )

**Comment by jacksongoode** [comment]
Well, I also imagine this thread might be useful to discuss whether it might make sense to attempt to make a maintained fork with attempting to merge existing PR's in?

**Comment by Insprill** [comment]
Since there hasn't been a response here, I've created a next branch in [my fork](https://github.com/Insprill/psst) with some of the current PRs merged in. Ideally, when/if jpochyla comes back, we can get that merged upstream and maybe create an org for it so development isn't reliant on one person.

**Reply by jacksongoode** [reply]
Cool!

**Comment by jpochyla** [comment]
Hi everybody! :( I'm sorry to disappoint you, I keep thinking there will be more time in the future, but it just never works out. I'd be definitely open to adding folks already familiar with the project as maintainers. It's cool that the project could live on, somehow. Thanks for being interested!

Now about some technical things. Part of the reason I've stopped is that currently, most of the complexity in the UI code is due to deficiencies of the Druid reactive architecture, which is getting completely re-done in the Xilem project. I'm eagerly following the Zulip (which I encourage everybody wanting to develop stuff with Druid to do), and Xilem looks amazing, but switching to Vello (nee piet-gpu) would lock out folks currently reliant on Piet for platform support (like the BSDs and such. I'm not sure how possible is to run Vello over there. Possibly it could be alright?). Anyway, Xilem will become usable in the next few months, probably. We could as well stick with the current architecture, of course, as it's working sort of alright.

Anyway, other major area, separate from the GUI, would be to focus on the internal crate and fix stuff like Spotify Connect support, that's a big one. We could as well decide to just scrap this all and switch to librespot, I'm not sure how is it nowadays, possible it progressed well since the time I've started to develop Psst.

**Comment by Insprill** [comment]
Hey, glad to see you're still around! 

I read the blog post about Xilem and it looks pretty promising. Considering a UI rework is on the roadmap, once it's ready for use that'd be a good time to take care of that. 

I don't know too much about librespot and they have little to no documentation that I could find, so I suppose we'd just have to play around with it a bit to see its current state. They do have Spotify Connect support and a few other niceties like lyrics which would save us a considerable amount of time, assuming everything else about it is stable and meets our needs.

If you're happy with the changes I made to my fork (All of them were done through PRs so it should be easy to look through), how do you suggest we merge the changes? None of them are too complex and could be opened as new PRs here with changes to existing ones moved over so we can merge them instead of closing.

**Reply by roderickvd** [reply]
I'm one of the librespot maintainers and if it means that it would get a bigger base of contributors, I am here to help.

**Reply by Insprill** [reply]
I just read #204, and I think joining forces could be highly beneficial to both projects. Shall we continue further discussion there?

**Comment by arch-btw** [comment]
Hi, I'm new to this project so take my words with a big grain of salt, but one of the reasons I moved to psst (besides it being amazing software, so thank you for that!) is also because it allows me to repeat a single track. 

Which has not been working in librespot since Jan 29, 2018. See https://github.com/librespot-org/librespot/issues/19

In turn this breaks repeat in the following projects:

Spotify-qt : https://github.com/kraxarn/spotify-qt/issues/206#issuecomment-1426726978
spotify-tui: https://github.com/Rigellute/spotify-tui/issues/1066#issuecomment-1407301811
spotify-player: https://github.com/aome510/spotify-player/issues/88
etc.

So my concern is that when relying on a single project that when something breaks all the other clients are affected too. Just my 2 cents.

**Comment by k0Yq** [comment]
Seems that the repository went back to being active, is the original maintainer back to business or there's a new maintainer?
